Sources/SwiftDocCUtilities/Action/Actions/MergeAction.swift

```diff
@@ -0,0 +1,65 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2024 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+@_spi(FileManagerProtocol) import SwiftDocC
+
+/// An action that merges a list of documentation archives into a combined archive.
+struct MergeAction: Action {
+    var archives: [URL]
+    var landingPageCatalog: URL?
+    var outputURL: URL
+    var fileManager: FileManagerProtocol
+
+    mutating func perform(logHandle: LogHandle) throws -> ActionResult {
+        guard let firstArchive = archives.first else {
+            // A validation warning should have already been raised in `Docc/Merge/InputAndOutputOptions/validate()`.
+            return ActionResult(didEncounterError: true, outputs: [])
+        }
+        
+        try? fileManager.removeItem(at: outputURL)
```

Comment:
Probably we should raise a validation error in `Merge` if the output directory already exists. Otherwise, the user might delete an entire directory structure by accident if they have a typo or other mistake in their command line.

Reply:
Fixed in https://github.com/apple/swift-docc/pull/821/commits/f98be845a62d75c7d8d019587e09793abd0ad12a with a new `testErrorWhenOutputDirectoryIsNotEmpty` test